File: Backend/Configs/strength_catalog.py
# Configs/strength_catalog.py

STRENGTH_EXERCISE_CATALOG = [
    # --- CORE ---
    {"id": "plank", "name_en": "Plank", "name_sk": "Plank", "target": "core", "equipment": ["none"]},
    {"id": "side_plank", "name_en": "Side Plank", "name_sk": "Bočný plank", "target": "core", "equipment": ["none"]},
    {"id": "abwheel_rollout", "name_en": "Ab Wheel Rollout", "name_sk": "Ab wheel rollout", "target": "core", "equipment": ["abwheel"]},
    {"id": "hanging_knee_raise", "name_en": "Hanging Knee Raise", "name_sk": "Zdvíhanie kolien vo vise", "target": "core", "equipment": ["pullup_bar"]},
    {"id": "cable_chop", "name_en": "Cable Woodchop", "name_sk": "Sťahovanie kladky zboku", "target": "core", "equipment": ["cable"]},
    {"id": "bird_dog", "name_en": "Bird-Dog", "name_sk": "Zdvihy na štyroch (Bird-dog)", "target": "core", "equipment": ["none"]},
    {"id": "dead_bug", "name_en": "Dead Bug", "name_sk": "Mŕtvy chrobák (Dead bug)", "target": "core", "equipment": ["none"]},

    # --- LOWER BODY (QUADS) ---
    {"id": "bodyweight_squat", "name_en": "Bodyweight Squat", "name_sk": "Drep s vlastnou váhou", "target": "lower_quad", "equipment": ["none"]},
    {"id": "barbell_back_squat", "name_en": "Barbell Back Squat", "name_sk": "Drep s veľkou činkou", "target": "lower_quad", "equipment": ["barbell"]},
    {"id": "leg_press_machine", "name_en": "Leg Press", "name_sk": "Leg press stroj", "target": "lower_quad", "equipment": ["machine"]},
    {"id": "split_squat", "name_en": "Split Squat", "name_sk": "Rozdelený drep (Split squat)", "target": "lower_quad", "equipment": ["none", "dumbbell"]},
    {"id": "goblet_squat", "name_en": "Goblet Squat", "name_sk": "Goblet drep", "target": "lower_quad", "equipment": ["dumbbell", "kettlebell"]},
    {"id": "box_stepup", "name_en": "Box Step-up", "name_sk": "Výstupy na debnu", "target": "lower_quad", "equipment": ["none", "dumbbell"]},

    # --- LOWER BODY (POSTERIOR) ---
    {"id": "glute_bridge_bodyweight", "name_en": "Glute Bridge", "name_sk": "Glute bridge", "target": "lower_posterior", "equipment": ["none"]},
    {"id": "romanian_deadlift_barbell", "name_en": "Romanian Deadlift", "name_sk": "Rumunský mŕtvy ťah (Barbell)", "target": "lower_posterior", "equipment": ["barbell"]},
    {"id": "romanian_deadlift_dumbbell", "name_en": "Dumbbell RDL", "name_sk": "Rumunský mŕtvy ťah (Dumbbell)", "target": "lower_posterior", "equipment": ["dumbbell"]},
    {"id": "single_leg_deadlift_band", "name_en": "Single Leg Deadlift", "name_sk": "Mŕtvy ťah na 1 nohe", "target": "lower_posterior", "equipment": ["none", "resistance_bands", "dumbbell"]},
    {"id": "hamstring_curl_machine", "name_en": "Hamstring Curl", "name_sk": "Zakopávanie stroj", "target": "lower_posterior", "equipment": ["machine"]},
    {"id": "hip_thrust_barbell", "name_en": "Barbell Hip Thrust", "name_sk": "Hip thrust s činkou", "target": "lower_posterior", "equipment": ["barbell"]},

    # --- UPPER PULL ---
    {"id": "bodyweight_row", "name_en": "Inverted Row", "name_sk": "Príťahy na hrazde (vodorovne)", "target": "upper_pull", "equipment": ["barbell", "trx"]},
    {"id": "trx_row", "name_en": "TRX Row", "name_sk": "TRX príťahy", "target": "upper_pull", "equipment": ["trx"]},
    {"id": "lat_pulldown_machine", "name_en": "Lat Pulldown", "name_sk": "Sťahovanie kladky na chrbát", "target": "upper_pull", "equipment": ["machine"]},
    {"id": "pullup_assisted", "name_en": "Assisted Pull-up", "name_sk": "Zhyby s dopomocou", "target": "upper_pull", "equipment": ["pullup_bar"]},
    {"id": "dumbbell_row", "name_en": "Single Arm Dumbbell Row", "name_sk": "Príťahy jednoručky v predklone", "target": "upper_pull", "equipment": ["dumbbell", "bench"]},

    # --- UPPER PUSH ---
    {"id": "pushup", "name_en": "Push-up", "name_sk": "Kľuk", "target": "upper_push", "equipment": ["none"]},
    {"id": "bench_press_barbell", "name_en": "Barbell Bench Press", "name_sk": "Tlak na lavičke", "target": "upper_push", "equipment": ["barbell", "bench"]},
    {"id": "incline_db_press", "name_en": "Incline Dumbbell Press", "name_sk": "Tlaky jednoručiek na šikmej lavičke", "target": "upper_push", "equipment": ["dumbbell", "bench"]},
    {"id": "shoulder_press_dumbbell", "name_en": "Dumbbell Shoulder Press", "name_sk": "Tlak jednoručkami nad hlavu", "target": "upper_push", "equipment": ["dumbbell"]},
    {"id": "dip_assisted", "name_en": "Assisted Dips", "name_sk": "Kľuky na bradlách s dopomocou", "target": "upper_push", "equipment": ["machine", "pullup_bar"]},
]

File: frontend/src/app/shared/constants/strengthCatalog.ts
// src/app/shared/constants/strengthCatalog.ts

export type ExerciseLangMap = {
  en: string;
  sk: string;
};

// Kľúč je exercise_id, hodnota je preklad. 
export const STRENGTH_CATALOG_FE: Record<string, ExerciseLangMap> = {
  // Core
  "plank": { en: "Plank", sk: "Plank" },
  "side_plank": { en: "Side Plank", sk: "Bočný plank" },
  "abwheel_rollout": { en: "Ab Wheel Rollout", sk: "Ab wheel rollout" },
  "hanging_knee_raise": { en: "Hanging Knee Raise", sk: "Zdvíhanie kolien vo vise" },
  "cable_chop": { en: "Cable Woodchop", sk: "Sťahovanie kladky zboku" },
  "bird_dog": { en: "Bird-Dog", sk: "Zdvihy na štyroch (Bird-dog)" },
  "dead_bug": { en: "Dead Bug", sk: "Mŕtvy chrobák (Dead bug)" },
  "russian_twist": { en: "Russian Twist", sk: "Ruský twist" },
  "mountain_climber": { en: "Mountain Climber", sk: "Horolezec (Mountain climber)" },
  "hollow_body_hold": { en: "Hollow Body Hold", sk: "Kolíska (Hollow hold)" },

  // Lower Quad
  "bodyweight_squat": { en: "Bodyweight Squat", sk: "Drep s vlastnou váhou" },
  "barbell_back_squat": { en: "Barbell Back Squat", sk: "Drep s veľkou činkou" },
  "leg_press_machine": { en: "Leg Press", sk: "Leg press stroj" },
  "split_squat": { en: "Split Squat", sk: "Rozdelený drep (Split squat)" },
  "goblet_squat": { en: "Goblet Squat", sk: "Goblet drep" },
  "box_stepup": { en: "Box Step-up", sk: "Výstupy na debnu" },
  "bulgarian_split_squat": { en: "Bulgarian Split Squat", sk: "Bulharský drep" },
  "walking_lunge": { en: "Walking Lunge", sk: "Kráčavé výpady" },
  "hack_squat_machine": { en: "Hack Squat", sk: "Hack drep stroj" },
  "leg_extension_machine": { en: "Leg Extension", sk: "Predkopávanie stroj" },

  // Lower Posterior
  "glute_bridge_bodyweight": { en: "Glute Bridge", sk: "Glute bridge (Dvíhanie panvy)" },
  "romanian_deadlift_barbell": { en: "Barbell RDL", sk: "Rumunský mŕtvy ťah (Činka)" },
  "romanian_deadlift_dumbbell": { en: "Dumbbell RDL", sk: "Rumunský mŕtvy ťah (Jednoručky)" },
  "single_leg_deadlift_band": { en: "Single Leg Deadlift", sk: "Mŕtvy ťah na 1 nohe" },
  "hamstring_curl_machine": { en: "Hamstring Curl", sk: "Zakopávanie stroj" },
  "hip_thrust_barbell": { en: "Barbell Hip Thrust", sk: "Hip thrust s činkou" },
  "kettlebell_swing": { en: "Kettlebell Swing", sk: "Kettlebell swing" },
  "conventional_deadlift": { en: "Conventional Deadlift", sk: "Klasický mŕtvy ťah" },
  "good_morning_barbell": { en: "Barbell Good Morning", sk: "Good morning s činkou" },
  "back_extension": { en: "Back Extension", sk: "Extenzie chrbta (Hyperextenzie)" },

  // Lower Calves
  "standing_calf_raise": { en: "Standing Calf Raise", sk: "Výpony v stoji" },
  "seated_calf_raise": { en: "Seated Calf Raise", sk: "Výpony v sede" },
  "single_leg_calf_raise": { en: "Single Leg Calf Raise", sk: "Výpony na jednej nohe" },
  "jump_rope": { en: "Jump Rope / Pogo Jumps", sk: "Švihadlo / Pogo výskoky" },
  "tibialis_raise": { en: "Tibialis Raise", sk: "Zdvíhanie špičiek (Tibialis)" },

  // Upper Pull
  "bodyweight_row": { en: "Inverted Row", sk: "Príťahy na hrazde (vodorovne)" },
  "trx_row": { en: "TRX Row", sk: "TRX príťahy" },
  "lat_pulldown_machine": { en: "Lat Pulldown", sk: "Sťahovanie kladky na chrbát" },
  "pullup_assisted": { en: "Assisted Pull-up", sk: "Zhyby s dopomocou" },
  "pullup_strict": { en: "Strict Pull-up", sk: "Zhyby (Príťahy nadhmatom)" },
  "dumbbell_row": { en: "Single Arm Dumbbell Row", sk: "Príťahy jednoručky v predklone" },
  "barbell_row": { en: "Barbell Bent-Over Row", sk: "Príťahy s veľkou činkou v predklone" },
  "seated_cable_row": { en: "Seated Cable Row", sk: "Príťahy na spodnej kladke v sede" },
  "face_pull": { en: "Face Pull", sk: "Face pull (Sťahovanie kladky k tvári)" },
  "chin_up": { en: "Chin-up", sk: "Zhyby podhmatom" },

  // Upper Push
  "pushup": { en: "Push-up", sk: "Kľuk" },
  "bench_press_barbell": { en: "Barbell Bench Press", sk: "Tlak na lavičke (Bench press)" },
  "incline_db_press": { en: "Incline Dumbbell Press", sk: "Tlaky jednoručiek na šikmej lavičke" },
  "shoulder_press_dumbbell": { en: "Dumbbell Shoulder Press", sk: "Tlak jednoručkami nad hlavu" },
  "dip_assisted": { en: "Assisted Dips", sk: "Kľuky na bradlách s dopomocou" },
  "dip_strict": { en: "Strict Dips", sk: "Kľuky na bradlách (Dipy)" },
  "overhead_press_barbell": { en: "Overhead Press", sk: "Tlak s veľkou činkou nad hlavu" },
  "push_press": { en: "Push Press", sk: "Push press (Tlak s dopomocou nôh)" },
  "pec_deck_fly": { en: "Pec Deck Fly", sk: "Peck Deck (Rozpažovanie stroj)" },
  "triceps_pushdown": { en: "Cable Triceps Pushdown", sk: "Sťahovanie kladky na triceps" },
};